firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -355,7 +373,8 @@ private InstallationTokenResult getValidAuthToken(PersistedFidEntry persistedFid
 
   private boolean isPersistedFidRegistered(PersistedFidEntry persistedFidEntry) {
     return persistedFidEntry != null
-        && persistedFidEntry.getRegistrationStatus() == RegistrationStatus.REGISTERED;
+        && (persistedFidEntry.getRegistrationStatus() == RegistrationStatus.REGISTERED
+            || persistedFidEntry.getRegistrationStatus() == RegistrationStatus.PENDING);
```

Comment:
This can be removed now, right?

Reply:
No. We need this check. With this new flow - both getId and getAuthToken need this. 

getAuthToken uses it with persistedFid  in REGISTERED state.
getId uses it with persistedFid  in PENDING state.